pkg/apis/core/validation/shoot.go

```diff
@@ -114,7 +114,7 @@ var (
 		core.ErrorConfigurationProblem,
 	)
 	// ForbiddenShootFinalizersOnCreation is a list of finalizers which are forbidden to be specified on Shoot creation.
-	ForbiddenShootFinalizersOnCreation = sets.New(
+	ForbiddenShootFinalizersOnCreation = sets.New[string](
```

Comment:
Why do we need this change?

Reply:
Might be because of some specific local tooling versions.
![image](https://github.com/user-attachments/assets/fb8b1422-0dff-47c8-a6ec-86cd534494e5)
Will recheck...

Reply:
still marked in red locally, but will leave out of this PR.